unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -4359,22 +4359,31 @@ float allowedError
                 return;
             }
 
-            var metadata = assetDb.prefabs.Where(p => p.GetComponent<SimObjPhysics>() != null).Select(
-                p => {
-                    var simObj = p.GetComponent<SimObjPhysics>();
-                    var bb = simObj.AxisAlignedBoundingBox;
-                    return new AssetMetadata() {
-                        id = simObj.gameObject.name,
-                        type = simObj.Type.ToString(),
-                        primaryProperty = simObj.PrimaryProperty.ToString(),
-                        secondaryProperties = simObj.SecondaryProperties.Select(s => s.ToString()).ToList(),
-                        boundingBox = new BoundingBox() {
-                            min = bb.center - bb.size / 2.0f,
-                            max = bb.center + bb.size / 2.0f
-                        }
-                    };
+            var metadata = new Dictionary<string, Dictionary<string, object>>();
+            foreach (GameObject p in assetDb.prefabs) {
+                if (p.GetComponent<SimObjPhysics>() == null) {
+                    continue;
+                }
+
+                var simObj = p.GetComponent<SimObjPhysics>();
+                var bb = simObj.AxisAlignedBoundingBox;
+
+                // TODO: no objects should have duplicate names.
+                if (metadata.ContainsKey(simObj.gameObject.name)) {
+                    continue;
                 }
-            ).ToList();
+
+                metadata.Add(simObj.gameObject.name, new Dictionary<string, object>() {
+                    ["objectType"] = simObj.Type.ToString(),
+                    ["primaryProperty"] = simObj.PrimaryProperty.ToString(),
+                    ["secondaryProperties"] = simObj.SecondaryProperties.Select(s => s.ToString()).ToList(),
+                    ["boundingBox"] = new BoundingBox() {
+                        min = bb.center - bb.size / 2.0f,
+                        max = bb.center + bb.size / 2.0f
+                    }
+                });
+            }
+
```

Comment:
Simpler way to do this and throws and exception if there are duplicates:

```
var metadata = assetDb.prefabs.Where(p => p.GetComponent<SimObjPhysics>() != null).ToDictionary(
                p => p.name,
                p => {
                    var simObj = p.GetComponent<SimObjPhysics>();
                    var bb = simObj.AxisAlignedBoundingBox;
                    return new AssetMetadata() {
                        id = simObj.gameObject.name,
                        type = simObj.Type.ToString(),
                        primaryProperty = simObj.PrimaryProperty.ToString(),
                        secondaryProperties = simObj.SecondaryProperties.Select(s => s.ToString()).ToList(),
                        boundingBox = new BoundingBox() {
                            min = bb.center - bb.size / 2.0f,
                            max = bb.center + bb.size / 2.0f
                        }
                    };
                }
            );
```

Reply:
Re https://github.com/allenai/ai2thor/pull/887/files#r708716485, I don't want to throw an exception right now, since it would make it unusable.

And, I feel like this makes it harder to maintain, since:
1. If the return type is ever changed, `AssetMetadata` must be updated.
2. It is weird for `id` to be in the `AssetMetadata` since the `id` it is now just the dict key.